src/lib.rs

```diff
@@ -762,6 +762,151 @@ pub trait Filesystem {
     }
 }
 
+#[derive(Debug, Hash, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Inode(u64);
+
+impl From<u64> for Inode {
+    fn from(ino: u64) -> Self {
+        Self(ino)
+    }
+}
+
+// we are missing most of the error codes provided by linux
+// if necessary, add the necessary.
+// https://man7.org/linux/man-pages/man3/errno.3.html
+pub enum Errno {
```

Comment:
This looks good, but I think we should use more descriptive error names, so instead of `EFBIG` something like `FileTooLarge`